### ISSUE TITLE
Core Text Renderer Not Accessible in VoiceOver

### ISSUE THREAD

**Issue by kolanos** [macvim-dev/macvim#1491]
**Is your feature request about something that is currently impossible or hard to do? Please describe the problem.**

Under the `Advanced` menu in settings for the checkbox `Use Core Text renderer` appears the message:

> (Deprecated) Deselect this option to use the legacy renderer, which is unsupported and will be removed in a future version.

However, at present, the Core Text renderer is not accessible in VoiceOver.

**Describe the solution you'd like**

Make Core Text renderer accessible in VoiceOver prior to the removal of the legacy renderer.

**Describe alternatives you've considered**

If the legacy renderer is removed prior to making Core Text accessible then I won't be able to use MacVim anymore as a blind user. And MacVim is great, so this would be a sad day indeed.

I'd have to use *gasp*  emacs? Or use vim in a terminal full time? Yuck.

**Comment by ychin** [comment]
This is certainly something I would like to support. Thanks for tagging this so I will keep this in mind.

Can you explain a little more about your use case? Are you completely relying on solely the macOS accessibility features to type on MacVim? Do you use other text editors and how do they compare (e.g. I know VSCode has a list of accessibility features). Do you use terminal Vim?

Just trying to understand more how you usually use MacVim here.

**Comment by kolanos** [comment]
@ychin I have used VS Code but have found many gaps in its accessibility. I know it has a Vim mode but everything around it gets in the way. Mac's built-in text editor, TextEdit, is a good example of what a text editor can be in VoiceOver. But it is not well suited for coding. Plus I have been a vim user going on 15 years, so old habits die hard.

As for terminals, Mac's built-in terminal in my experience has been more accessible than alternatives like iterm. I have used vim and neovim in a terminal emulator, and as a fallback this is adequate. I have tried neovim GUIs like neoviee and they are completely inaccessible by VoiceOver. Similar to the Core Text renderer today.

I just recently upgraded to Sonoma and there is (sadly) a new (what I presume to be a) VoiceOver bug in MacVim with Core Text renderer disabled. When I try to type in the command  line VoiceOver reads the first `:` character and then just says `"space" for every character after that. I need to read the entire line to see what is there (VO+L). So even VoiceOver itself sees regressions in this regard.

This feature is likely a big ask. So a possible alternative is a legacy version of MacVim with the legacy renderer intact.

**Comment by sslotay** [comment]
I am a blind user on macos using only voiceover. I have fell in love with vim motion but as described by another comment here, voiceover has been having severe issues with terminal and cursor movements lately so I too have to use vo+l after jk in terminal vim to know the actual line I am on, which is completely unproductive and 100% ruins the vim experience I was starting to like. (on the latest stable release of macos sequoia, 15.3 as of writing). I've tried to install macvim via homebrew but it's completely inaccessible. It has been described in great detail here by myself and other users. https://www.applevis.com/comment/180698
I would like to use vim as a blind user , and if macvim can be the solution in an accessible way I'd be very grateful.

**Comment by kolanos** [comment]
@sslotay Disabling Advanced > Use Core Text renderer and putting MacVim into full screen mode seems to help with the Sonoma VoiceOver issues. Although I have a heavily customized `.vimrc` so I'm not sure if other things in my config are also helping.

Another thing I notice is that on first MacVim load VoiceOver has all kinds of issues. But If I `CTRL+Right` (`CTRL` and the right arrow key, in full screen mode) to leave the current full screen view and then `CTRl+Left` to get back to it MacVim no longer has the same VoiceOver issues.

**Comment by ychin** [comment]
Let me play around with VoiceOver a little bit. Core Text renderer is deprecated and may be ripped out so it does pain me to hear that there's still some use for it. There are definitely things we could do to improve the situation here and make MacVim support VoiceOver natively. For the current release (r181) I'm working on cleaning up the new tabs functionality but after that I do want to do more cleanups and misc support so I could look into this.

When I said "Can you explain a little more about your use case", I guess my question is, how much information do you want in VoiceOver? Do you want it to be able to just read the entire line? Or be able to tell you syntactical highlighting, where the command line is, and whatnot? Or I guess from my understanding *any* improvement is better than the current state.
